docs/source/add_new_model.md

```diff
@@ -0,0 +1,573 @@
+```
+
+In addition, you should also print out the names of both weights to make sure they match, *e.g.*
+
+```python
+logger.info(f"Initialize PyTorch weight {layer_name} from {pretrained_weight.name}")
+```
+
+If either the shape or the name doesn't match, you probably assigned the wrong 
+checkpoint weight to a randomly initialized layer of the 🤗Transformers implementation. 
+
+An incorrect shape is most likeliy due to an incorrect setting of the config parameters in `BrandNewBertConfig()` that do not exactly match those that were used for the checkpoint you want to convert. However, it could also be that PyTorch's implementation of a layer
+requires the weight to be transposed beforehand.
+
+Finally, you should also check that **all** required weights are initialized and print
+out all checkpoint weights that were not used for initialization to make sure the model is correctly converted.
+It is completely normal, that the conversion trials fail with either a wrong shape statement or wrong name assignment.
+This is most likely because either you used incorrect parameters in `BrandNewBertConfig()`, have a wrong architecture 
+in the 🤗Transformers implementation, you have a bug in the `init()` functions of one of the components of 
+the 🤗Transformers implementation or you need to transpose one of the checkpoint weights.
+
+This step should be iterated with the previous step until all weights of the checkpoint are correctly loaded in the Transformers model.
+Having correctly loaded the checkpoint into the 🤗Transformers implementation, you can then save the model under a folder of your choice `/path/to/converted/checkpoint/folder` that should include both a `pytorch_model.bin` file and a `config.json` file.
+
+#### Implement the forward pass
+
+Having managed to correctly load the pretrained weights into the 🤗Transformers implementation, you should now make sure that the forward pass is correctly implemented. In *Get familiar with the original repository.*, you have already created a script
+that runs a forward pass of the model using the original repository. Now you should write an analogous script using the 
+🤗Transformers implementation instead of the original one. It should look as follows:
+
+```python
+model = brand_new_bertModel.from_pretrained(/path/to/converted/checkpoint/folder)
+input_ids = ... # the exact same vector of input ids in PyTorch as those used in the *Get familiar with the original repository.* section
+output = model(input_ids).last_hidden_states
+```
+
+It is very likely that the 🤗Transformers implementation and the original model implementation don't give the exact same output the very first time or that the forward pass throws an error. Don't be disappointed - it's expected! First, you should make sure that the forward pass doesn't throw any errors. It often happens that the wrong dimensions are used leading to a `Dimensionality mismatch` error or that the wrong data type object is used,
```

Comment:
This part is great! It got me thinking, it could be nice to do a kind of "FAQ", where the headers are the errors that can happen. 

We can fill it up as we encounter them. What do you think? (Potential v2, not necessarily for this version)